src/test/java/org/example/FestivalGateUnitTest.java

```diff
@@ -0,0 +1,45 @@
+package org.example;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class FestivalGateUnitTest {
+
+    private final FestivalGate festivalGate = new FestivalGate();
+
+    @Test
+    void getValidTickets() {
+
+        festivalGate.getValidTickets().add(TicketType.ONEDAYVIP);
+        festivalGate.getValidTickets().add(TicketType.ONEDAY);
+        assertEquals(2, festivalGate.getValidTickets().size(), "The ticket queue should contain exactly 2 tickets.");
+        assertTrue(festivalGate.getValidTickets().contains(TicketType.ONEDAYVIP));
+        assertTrue(festivalGate.getValidTickets().contains(TicketType.ONEDAY));
+    }
+
+    @Test
+    void getValidTickets_NullQueue() {
```

Comment:
let's keep the usual java naming convention, avoid underscore
```suggestion
    void getValidTicketsNullQueue() {
```